apollo-ios/Sources/Apollo/FieldSelectionCollector.swift

```diff
@@ -27,6 +29,10 @@ struct FieldSelectionGrouping: Sequence {
     fulfilledFragments.insert(ObjectIdentifier(type))
   }
 
+  mutating func addDeferredFragment<T: SelectionSet>(_ type: T.Type) {
```

Comment:
I don't believe this is possible because the directive definition doesn't allow `@defer` to be used on fields. It's only valid on inline and named fragments - `directive @defer(label: String, if: Boolean! = true) on FRAGMENT_SPREAD | INLINE_FRAGMENT`.